spec/liquid_validations_spec.rb

```diff
@@ -62,4 +66,101 @@ class Mixin < ActiveRecord::Base
+     @mixin.errors.full_messages.any? { |e| e == "You must supply {% josh_is_awesome %} in your content"}.must_equal false
+   end
+
+   it 'must be valid when tag is less than or equal max count' do
+     @mixin.content = '{% josh_is_awesome %} {% josh_is_awesome %}'
+     @mixin.valid?
+     @mixin.errors.full_messages.any? { |e| e == "You must supply {% josh_is_awesome %} in your content"}.must_equal false
+   end
+
+   it 'must be invalid when tag is greater than max count' do
+     @mixin.content = '{% josh_is_awesome %} {% josh_is_awesome %} {% josh_is_awesome %}'
+     @mixin.valid?
+     @mixin.errors.full_messages.any? { |e| e == "content must not have more than max tags 2"}.must_equal true
+   end
+
+   it 'must be valid when using more complex tag' do
+     @mixin.content = "{% josh_is_awesome foobar, data-required='true' %}"
+     @mixin.valid?
+     @mixin.errors.full_messages.any? { |e| e == "You must supply {% josh_is_awesome %} in your content"}.must_equal false
+   end
+
+   it 'must be invalid when using more complex tag' do
+     @mixin.content = "{% josh_is_awesome foobar, data-required='true' %}"
+     @mixin.valid?
+     @mixin.errors.full_messages.any? { |e| e == "You must supply {% josh_is_awesome %} in your content"}.must_equal false
+   end
+
+   it 'must be invalid when using tag like { josh_is_awesome }' do
+     @mixin.content = "{ josh_is_awesome }"
+     @mixin.valid?
+     @mixin.errors.full_messages.any? { |e| e == "You must supply {% josh_is_awesome %} in your content"}.must_equal true
+   end
+
+   it 'must be invalid when using tag like {% josh_is_awesome }' do
+     @mixin.content = "{% josh_is_awesome }"
+     @mixin.valid?
+     @mixin.errors.full_messages.any? { |e| e == "You must supply {% josh_is_awesome %} in your content"}.must_equal true
+   end
+
+   it 'must be invalid when using tag like { josh_is_awesome %}' do
+     @mixin.content = "{ josh_is_awesome %}"
+     @mixin.valid?
+     @mixin.errors.full_messages.any? { |e| e == "You must supply {% josh_is_awesome %} in your content"}.must_equal true
+   end
+
+   it 'must be invalid when using tag like {%% josh_is_awesome %%}' do
+     @mixin.content = "{%% josh_is_awesome %%}"
+     @mixin.valid?
+     @mixin.errors.full_messages.any? { |e| e == "You must supply {% josh_is_awesome %} in your content"}.must_equal true
+   end
+
+   it 'must be invalid when using tag like %{ josh_is_awesome }%' do
+     @mixin.content = "%{ josh_is_awesome }%"
+     @mixin.valid?
+     @mixin.errors.full_messages.any? { |e| e == "You must supply {% josh_is_awesome %} in your content"}.must_equal true
+   end
+
+   it 'must be invalid when using tag like empty content' do
+     @mixin.valid?
+     @mixin.errors.full_messages.any? { |e| e == "You must supply {% josh_is_awesome %} in your content"}.must_equal true
+   end
+
+   it 'must be invalid when tag is greater than max count' do
+     @mixin.content = '{% josh_is_awesome %} {% josh_is_awesome %} {% josh_is_'
```

Comment:
the third tag here is incomplete: {% josh_is_

also this test should not fail with the current content. So add a second test with correct tag, and keep this test but make sure it is valid.

Reply:
Yes with this test it should be valid I will add another test with correct tag